suites/apis/rasDRSTest.js

```diff
@@ -0,0 +1,248 @@
+  }
+
+  if (process.env.clientID === undefined) {
+    throw new Error('ERROR: ClientID is not defined. Please declare the "clientID" environment variable and try again. Aborting test...');
+  } else if (process.env.secretID === undefined) {
+    throw new Error('ERROR: SecretID is not defined. Please declare the "secretID" environment variable and try again. Aborting test...');
+  }
+
+  // adding clientID and secretID to the cache
+  I.cache.clientID = process.env.clientID;
+  I.cache.clientSecret = process.env.secretID;
+
+  // getting the access_token for the test user
+  I.cache.ACCESS_TOKEN = await bash.runCommand('gen3 api access-token cdis.autotest@gmail.com');
+  console.log(`Access_Token: ${I.cache.ACCESS_TOKEN}`);
+  // upload new indexdFile
+  const uploadResp = await I.sendPostRequest(
+    `https://${TARGET_ENVIRONMENT}/index/index`,
+    indexdFile.fileToUpload,
+    assembleCustomHeaders(I.cache.ACCESS_TOKEN),
+  );
+  // // if the authz resource path is hardcoded, you run this to fetch the did of file with hardcoded authz path
+  // // get indexd record for presignedurl request
+  // const getIndexRecord = await I.sendGetRequest(
+  //   `https://${TARGET_ENVIRONMENT}/index/index?authz=${authZ}`,
+  // );
+  I.cache.indexdRecord = uploadResp.data.did;
+  I.cache.indexdRev = uploadResp.data.rev;
+  console.log(`Indexd Record DID: ${I.cache.indexdRecord}`);
+});
+
+AfterSuite(async ({ I }) => {
+  // revoke the access for the next run
+  console.log('Revoking the access ..');
+  const revokeData = queryString.stringify({
+    token_type_hint: 'access_token',
+    token: `${I.cache.rasToken}`,
+    client_id: `${I.cache.clientID}`,
+    client_secret: `${I.cache.clientSecret}`,
+    scope: `${scope}`,
+  });
+
+  const revokeReq = await I.sendPostRequest(
+    `https://${rasServerURL}/auth/oauth/v2/token/revoke`,
+    revokeData,
+    {
+      'Content-Type': 'application/x-www-form-urlencoded',
+    },
+  );
+  if (revokeReq.status === 200) {
+    console.log('The access has been revoked');
+  }
+
+  // deleting the indexd record that was created in BeforeSuite
+  console.log('Deleting the indexd record ..');
+  const deleteRecordReq = await I.sendDeleteRequest(
+    `https://${TARGET_ENVIRONMENT}/index/index/${I.cache.indexdRecord}?rev=${I.cache.indexdRev}`,
+    {
+      Authorization: `Bearer ${I.cache.ACCESS_TOKEN}`,
+    },
+  );
+  if (deleteRecordReq.status === 200) {
+    console.log('Indexd record is deleted');
+  }
+});
+
+async function getTokens(I) {
+  console.log('Getting the Auth Code ...');
+  I.amOnPage(`https://${rasServerURL}/auth/oauth/v2/authorize?response_type=code&client_id=${I.cache.clientID}&redirect_uri=https://qa-dcp.planx-pla.net/user/login/ras/callback&scope=${scope}&idp=ras`);
+  // GET auth/oauth/v2/authorize to get the rasAuthCode
+  await sleepMS(3000);
```

Comment:
Is this to wait for the login form to load? If so, would there be a way to instead wait until either the login form (or entire) DOM loads? That way, if it takes longer than 3 seconds to load, the test will still be able to find the element.

Reply:
we can! we can check if any element (eg. Sign-in button) loads instead of waiting for a specific amount of time. I would have do something like `I.waitforElement`.